### PR TITLE
Add generic info to Map & Array annotation

### PR DESCRIPTION
**Pull request by tisonkun**
Signed-off-by: tison <wander4096@gmail.com>

straightforward

### CODE REVIEW

**Comment by rubenada** [comment]
This patch could be merged, but there is no associated Jira. Since it is a small change, perhaps we could make an exception, what do you think @zabetak ?

**Comment by zabetak** [comment]
@rubenada changes with small impact to end-users can go in without JIRA ;)

**Comment by rubenada** [comment]
Thanks for the confirmation @zabetak

**Comment by amaliujia** [comment]
Just out of curiosity: what is the benefit to have generic info?

File: core/src/main/java/org/apache/calcite/adapter/java/Map.java [hunk 1]
```diff
@@ -16,23 +16,22 @@
...
 /**
  * Annotation that indicates that a field is a map type.
  */
-@Target({FIELD })
+@Target(ElementType.FIELD)
 @Retention(RetentionPolicy.RUNTIME)
 public @interface Map {
   /** Key type. */
-  Class key();
+  Class<?> key();
```
Comment (amaliujia) [inline core/src/main/java/org/apache/calcite/adapter/java/Map.java:31]: What is changed after having `Class<?>`?

**Comment by tisonkun** [comment]
@rubenada @zabetak @amaliujia  thanks for your time!

> Just out of curiosity: what is the benefit to have generic info?

@amaliujia I think adding generic parameter when there should be one is a general best practice. It is useful as linter in this case.

I created this pull request on a day reading the source code and linter complain. The community may or may not reject such pr and require to be included in a larger one, but I prefer we accept valid contributions at any level. If you turn a blind eye to generic parameter missing, it is possible we run into a case where Apache Hudi is, that it is hard to fix missing generic parameters that really hurt.

**Comment by rubenada** [comment]
Thanks @tisonkun for the patch and for the detailed explanation!

**Pull request merged**